projects/vanilla/src/app/configurators/color-picker-control.component.ts

```diff
@@ -12,20 +12,36 @@ import { ConfiguratorContext } from "@sinequa/ngx-ui-builder";
     [style.background]="color"/>
   `
 })
-export class ColorPickerControlComponent {
+export class ColorPickerControlComponent implements OnChanges {
   @Input() context: ConfiguratorContext;
   @Input() property: string;
   @Input() label?: string;
 
+  _path: string[];
+
   get color() {
-    return this.context.config[this.property];
+    let val = this.context.config;
+    for(let p of this._path) {
+      val = val[p];
+    }
+    return <string><unknown>val;
```

Comment:
Yes... this is bad, but I didn't find a way around it: Typescript thinks the type of `val` is `ComponentConfig` because of line 23. So I have to cast val as a `string` to force the correct return type...